plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

```diff
@@ -279,8 +280,13 @@ private DataFileInfo rewriteParquetFile(Path path, ImmutableLongBitmapDataProvid
                     page = page.getPositions(retained, 0, retainedCount);
                 }
 
+                rowsWritten += page.getPositionCount();
                 fileWriter.appendRows(page);
             }
+            if (rowsWritten == 0) {
```

Comment:
`DeltaLakeWriter` already has a `rowCount`. We could add a getter for that.
```java
if (fileWriter.getRowCount() == 0) {

Reply:
Added `DeltalLakeFileWriter.getRowCount()`, used it here and removed `rowsWritten`.